### ISSUE TITLE
BUG: to_postgis uses wrong SRID for CRS with ESRI authority

### ISSUE THREAD

**Issue by ryan-salo** [geopandas/geopandas#2414]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of geopandas.

- [x] (optional) I have confirmed this bug exists on the main branch of geopandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import geopandas as gpd
from shapely.geometry import Point
from sqlalchemy import create_engine
from pyproj import CRS


engine = create_engine("postgresql://user:password@SERVERIP:5432/db")
gdf = gpd.GeoDataFrame(
    {
        "field1": [42],
        "geometry": [Point(-962003.27949604, 286498.51751122)]
    },
    crs=CRS.from_string("ESRI:102003")
)

gdf.to_postgis(
    "testing_table",
    con=engine,
    if_exists="append",
    index=False,
)

new_df = gpd.read_postgis(
    "SELECT * FROM testing_table",
    engine,
    geom_col="geometry"
)

assert new_df.crs.equals(aea)
```

#### Problem description

It looks like [geopandas uses](https://github.com/geopandas/geopandas/blob/eae47910b7e9675d6d3c6376a649f64448b1fcdc/geopandas/io/sql.py#L252)  `to_epsg` with a `min_confidence` value. Perhaps using something like `to_authority` would be better?

```python
def _get_srid_from_crs(gdf):
    """
    Get EPSG code from CRS if available. If not, return -1.
    """

    # Use geoalchemy2 default for srid
    # Note: undefined srid in PostGIS is 0
    srid = -1
    warning_msg = (
        "Could not parse CRS from the GeoDataFrame. "
        + "Inserting data without defined CRS.",
    )
    if gdf.crs is not None:
        try:
            authority, srid = gdf.crs.to_authority(min_confidence=25)
            srid = int(srid)
            if srid is None:
                srid = -1
                warnings.warn(warning_msg, UserWarning, stacklevel=2)
        except Exception:
            warnings.warn(warning_msg, UserWarning, stacklevel=2)
    return srid
```

The SRID `102003` does exist in the `spatial_ref_sys` table.

#### Output of ``geopandas.show_versions()``

<details>

SYSTEM INFO
-----------
python     : 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:22:55)  [GCC 10.3.0]
executable : /home/user/miniconda3/envs/gpd-env/bin/python
machine    : Linux-5.16.15-76051615-generic-x86_64-with-glibc2.34

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : 3.10.2
GEOS lib   : /home/user/miniconda3/envs/gpd-env/lib/libgeos_c.so
GDAL       : 3.4.1
GDAL data dir: /home/user/miniconda3/envs/gpd-env/share/gdal
PROJ       : 8.2.1
PROJ data dir: /home/user/miniconda3/envs/gpd-env/share/proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.10.2
pandas     : 1.4.2
fiona      : 1.8.21
numpy      : 1.22.3
shapely    : 1.8.0
rtree      : 1.0.0
pyproj     : 3.3.0
matplotlib : 3.5.1
mapclassify: 2.4.3
geopy      : None
psycopg2   : 2.9.3 (dt dec pq3 ext lo64)
geoalchemy2: None
pyarrow    : None
pygeos     : None

</details>

**Comment by snowman2** [comment]
I think this is most likely due to the fact that PostGIS currently has an outdated/limited `spatial_ref_sys` table compared to the PROJ database.

**Comment by martinfleis** [comment]
Can we say that it needs to be fixed on the PostGIS side then?

**Comment by martinfleis** [comment]
Closed by #2415.
